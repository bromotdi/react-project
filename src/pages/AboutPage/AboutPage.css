.aboutPage {
  margin-bottom: 3rem;
  padding: 0 1rem;
}

.about-page__greeting {
    font-size: 22px;
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 50%;
}

.aboutPage__img {
   display: block;
   margin-left: auto;
   margin-right: auto;
   width: 50%;
}

.aboutPage__text {
  font-size: 1.4rem;
  font-weight: 400;
  color: #124d53;
  margin: 2rem 0;
}

@media (max-width: 400px) {
  .aboutPage {
    font-size: 0.8rem;
    font-weight: 300;
    color: #124d53;
  }
}
