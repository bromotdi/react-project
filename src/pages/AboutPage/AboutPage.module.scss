@import '../../scss/mixins';
@import '../../scss/variables';

.aboutPage {
  margin-bottom: 3rem;
  padding: 0 1rem;

  &__title {
    @include font(3.6rem, 700, $text-color-main);
    margin: 0 0 2rem 4rem;
  }

  &__text {
    @include font(2.4rem, 400, $text-color-main);
  }
}
