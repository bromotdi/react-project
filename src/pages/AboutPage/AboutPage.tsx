import React from 'react';
import style from './AboutPage.module.scss';

export class AboutPage extends React.Component {
  render() {
    return (
      <div className={style.aboutPage}>
        <h1 className={style.aboutPage__title}>About</h1>
        <p className={style.aboutPage__text}>
          Attack on Titan is a Japanese dark fantasy anime television series, adapted from the manga
          series of the same name by Hajime Isayama, that premiered on April 7, 2013. It has aired
          on NHK General TV in Japan, and Aniplus Asia in various Asia-Pacific countries. In the In
          the United States and Canada, the series has been streamed on Crunchyroll, Funimation,
          Netflix, Amazon Prime Video, and Hulu. Attack on Titan has also aired on Adult Swim&apos;s
          Toonami programming block in the U.S.
        </p>
        <p className={style.aboutPage__text}>
          Set in a post-apocalyptic world where the remains of humanity live behind walls protecting
          them from giant humanoid Titans, Attack on Titan follows protagonist Eren Yeager, along
          with friends Mikasa Ackerman and Armin Arlert. When a Colossal Titan breaches the wall of
          their hometown, Titans destroy the city and eat Eren&apos;s mother. Vowing vengeance, Eren
          joins the elite Survey Corps, a group of soldiers who fight against Titans. It chronicles
          Eren&apos;s journey with the Survey Corps as they fight against the Titans while
          investigating their origin and history.
        </p>
        <p className={style.aboutPage__text}>
          Since its debut in 2013, Attack on Titan has received widespread critical acclaim and
          multiple accolades. Critics and audiences have praised the show for its story line,
          animation, action sequences, characters, voice acting (both original and dubbed),
          soundtrack, and dark themes.
        </p>
      </div>
    );
  }
}
