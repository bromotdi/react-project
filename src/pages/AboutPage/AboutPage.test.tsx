import { render, screen } from '@testing-library/react';
import { AboutPage } from './AboutPage';

describe('AboutPage', () => {
  it('AboutPage snapshot', () => {
    const container = render(<AboutPage />);
    expect(container).toMatchSnapshot();
  });

  it('render AboutPage elements', () => {
    const { container } = render(<AboutPage />);
    expect(screen.getByText(/About/i)).toBeInTheDocument();
    expect(container.querySelectorAll('p').length).toBe(3);
  });
});
