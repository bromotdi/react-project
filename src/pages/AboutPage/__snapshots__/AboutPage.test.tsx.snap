// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage AboutPage snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="aboutPage"
      >
        <h1
          class="aboutPage__title"
        >
          About
        </h1>
        <p
          class="aboutPage__text"
        >
          Attack on Titan is a Japanese dark fantasy anime television series, adapted from the manga series of the same name by Hajime Isayama, that premiered on April 7, 2013. It has aired on NHK General TV in Japan, and Aniplus Asia in various Asia-Pacific countries. In the In the United States and Canada, the series has been streamed on Crunchyroll, Funimation, Netflix, Amazon Prime Video, and Hulu. Attack on Titan has also aired on Adult Swim's Toonami programming block in the U.S.
        </p>
        <p
          class="aboutPage__text"
        >
          Set in a post-apocalyptic world where the remains of humanity live behind walls protecting them from giant humanoid Titans, Attack on Titan follows protagonist Eren Yeager, along with friends Mikasa Ackerman and Armin Arlert. When a Colossal Titan breaches the wall of their hometown, Titans destroy the city and eat Eren's mother. Vowing vengeance, Eren joins the elite Survey Corps, a group of soldiers who fight against Titans. It chronicles Eren's journey with the Survey Corps as they fight against the Titans while investigating their origin and history.
        </p>
        <p
          class="aboutPage__text"
        >
          Since its debut in 2013, Attack on Titan has received widespread critical acclaim and multiple accolades. Critics and audiences have praised the show for its story line, animation, action sequences, characters, voice acting (both original and dubbed), soundtrack, and dark themes.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="aboutPage"
    >
      <h1
        class="aboutPage__title"
      >
        About
      </h1>
      <p
        class="aboutPage__text"
      >
        Attack on Titan is a Japanese dark fantasy anime television series, adapted from the manga series of the same name by Hajime Isayama, that premiered on April 7, 2013. It has aired on NHK General TV in Japan, and Aniplus Asia in various Asia-Pacific countries. In the In the United States and Canada, the series has been streamed on Crunchyroll, Funimation, Netflix, Amazon Prime Video, and Hulu. Attack on Titan has also aired on Adult Swim's Toonami programming block in the U.S.
      </p>
      <p
        class="aboutPage__text"
      >
        Set in a post-apocalyptic world where the remains of humanity live behind walls protecting them from giant humanoid Titans, Attack on Titan follows protagonist Eren Yeager, along with friends Mikasa Ackerman and Armin Arlert. When a Colossal Titan breaches the wall of their hometown, Titans destroy the city and eat Eren's mother. Vowing vengeance, Eren joins the elite Survey Corps, a group of soldiers who fight against Titans. It chronicles Eren's journey with the Survey Corps as they fight against the Titans while investigating their origin and history.
      </p>
      <p
        class="aboutPage__text"
      >
        Since its debut in 2013, Attack on Titan has received widespread critical acclaim and multiple accolades. Critics and audiences have praised the show for its story line, animation, action sequences, characters, voice acting (both original and dubbed), soundtrack, and dark themes.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
